File: ruby/62.unique-paths.rb
# @param {Integer} m
# @param {Integer} n
# @return {Integer}
def unique_paths(m, n)
  factorial(m + n - 2) / (factorial(m - 1) * factorial(n - 1))
end

def factorial(n)
  (1..n).inject(:*)
end

File: rust/src/62.unique-paths.rs
// use crate::Solution;

impl Solution {
    pub fn unique_paths(m: i32, n: i32) -> i32 {
        fn factorial(n: i32) -> i32 {
            (1..=n).product()
        }

        factorial(m + n - 2) / (factorial(m - 1) * factorial(n - 1))
    }
}
